types/type_map.go

```diff
@@ -99,3 +103,16 @@ func (t typeMap) Validate() error {
 	}
 	return t.value.Validate()
 }
+
+func genName(baseName string, key bool) string {
+	postfix := "value"
+	if key {
+		postfix = "key"
+	}
+	digest := md5.New()
+	n, err := digest.Write([]byte(baseName + postfix))
```

Comment:
it can't be the same, bc of names should be unique on first level, on every nested level previous name used to calc digest, so it also will be unique.